File: ruby_tutorials/hello_world_consumer.rb
require "amqp-client"
require "dotenv/load"

# Grab lavinmq url from .env file
lavinmq_url = ENV['CLOUDAMQP_URL']

# Opens and establishes a connection
connection = AMQP::Client.new(lavinmq_url).connect

# Open a channel
channel = connection.channel
puts "[✅] Connection over channel established"
puts "[❎] Waiting for messages. To exit press CTRL+C "

# Create a queue
queue = channel.queue_declare("hello_world")

counter = 0
# Subscribe to the queue
channel.basic_consume("hello_world") do |msg|
  counter += 1
  # Add logic to handle the message here...
  puts "[📤] Message received [#{counter}]: #{msg.body}"
end

# Close the connection when the script exits
at_exit do 
  client.stop
  puts "[❎] Connection closed"
end

# Keep the consumer running
sleep

File: deno_tutorials/hello_world_consumer.ts
import "https://deno.land/std@0.192.0/dotenv/load.ts"
import { connect } from "./deps.ts"

const hostname = Deno.env.get("CLOUDAMQP_URL") || "127.0.0.1"
const queue = Deno.env.get("CLOUDAMQP_QUEUE") || "deno.amqp.hello_world"

const connection = await connect({ hostname })
const channel = await connection.openChannel()

await channel.declareQueue({ queue })

await channel.consume(
  { queue },
  async (args, props, data) => {
    console.log(JSON.stringify(args))
    console.log(JSON.stringify(props))
    console.log(new TextDecoder().decode(data))
    console.log("=== 3< ===")

    await channel.ack({ deliveryTag: args.deliveryTag })
  },
);

console.log("[✅] Connection over channel established")
console.log("[❎] Waiting for messages. To exit press CTRL+C ")

const stopConsumer = async () => {
  await channel.close()
  await connection.close()
  
  console.log("[❎] Connection closed")
}

Deno.addSignalListener("SIGINT", stopConsumer)
Deno.addSignalListener("SIGTERM", stopConsumer)